File: services/hr-core/app/services/llm_client.py
import httpx

from app.core.config import LLM_BASE_URL, LLM_TIMEOUT_SECONDS
from app.schemas.screening import ScreeningRequest, ScreeningResult


class LLMClient:
    def __init__(self, base_url: str = LLM_BASE_URL, timeout_seconds: float = LLM_TIMEOUT_SECONDS) -> None:
        self.base_url = base_url.rstrip("/")
        self.timeout_seconds = timeout_seconds

    def screen_resume(self, payload: ScreeningRequest) -> ScreeningResult:
        with httpx.Client(timeout=self.timeout_seconds) as client:
            response = client.post(f"{self.base_url}/v1/screen", json=payload.model_dump())
            response.raise_for_status()
            return ScreeningResult(**response.json())

File: services/hr-core/src/services/llmClient.ts
import { z } from "zod";
import type { AppConfig } from "../config/env.js";
import type { ScreeningRequest, ScreeningResult } from "../domain/types.js";

const screeningResultSchema = z.object({
  clarifying_questions: z.array(z.string()),
  score: z.number(),
  score_reasons: z.array(z.string()),
  risks_to_clarify: z.array(z.string())
});

export interface LLMClient {
  screenResume(payload: ScreeningRequest): Promise<ScreeningResult>;
}

export class HttpLLMClient implements LLMClient {
  constructor(private readonly config: AppConfig) {}

  async screenResume(payload: ScreeningRequest): Promise<ScreeningResult> {
    const response = await fetch(`${this.config.llmBaseUrl.replace(/\/$/, "")}/v1/screen`, {
      method: "POST",
      headers: { "content-type": "application/json" },
      body: JSON.stringify({
        resume_text: payload.resumeText,
        mandatory_requirements: payload.mandatoryRequirements,
        optional_requirements: payload.optionalRequirements
      })
    });

    if (!response.ok) {
      throw new Error(`LLM screening failed with status ${response.status}`);
    }

    const parsed = screeningResultSchema.parse(await response.json());
    return {
      clarifyingQuestions: parsed.clarifying_questions,
      score: parsed.score,
      scoreReasons: parsed.score_reasons,
      risksToClarify: parsed.risks_to_clarify
    };
  }
}
